apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloCacheInterceptor.kt

```diff
@@ -100,19 +103,23 @@ class ApolloCacheInterceptor(
         && !request.cacheHeaders.hasHeader(ApolloCacheHeaders.STORE_PARTIAL_RESPONSES)) {
       return emptySet()
     }
-    val records = networkResponse.cacheRecords.orNull()?.let { records ->
-      val result: MutableList<Record> = ArrayList(records.size)
-      for (record in records) {
-        result.add(record.toBuilder().mutationId(request.uniqueId).build())
-      }
-      result
+
+    val rawRecords = networkResponse.parsedResponse.get()?.data?.let {
+      (request.operation as Operation<Operation.Data>)
+          .normalize(it, customScalarAdapters, apolloStore.networkResponseNormalizer() as ResponseNormalizer<Map<String, Any>?>)
+    }
+    val mergeRecords = rawRecords?.map {
+      it.toBuilder().mutationId(request.uniqueId).build()
```

Comment:
Do everything in one step ? 

```
    val rawRecords = networkResponse.parsedResponse.get()?.data?.let {
      (request.operation as Operation<Operation.Data>)
          .normalize(it, customScalarAdapters, apolloStore.networkResponseNormalizer() as ResponseNormalizer<Map<String, Any>?>)
    }?.map { it.toBuilder().mutationId(request.uniqueId).build() }
```

Not that I mind having 2 steps but `mergeRecords` is confusing as "merge" usually refers to the process of merging several records together.

Reply:
Oh yeah I remember doing that cause the extension `dependentkeys()` was expecting a `Set<Record>`.
Let me check if it's not used anywhere else anymore, I can just refactor them to use a `List<Record>` instead, it would be simpler indeed.